src/PKSim.Matlab/MatlabPopulationFactory.cs

```diff
@@ -1,41 +1,73 @@
+      IParameterValueCache CreatePopulation(PopulationSettings matlabPopulationSettings, IEnumerable<MoleculeOntogeny> moleculeOntogenies);
    }
 
    public class MatlabPopulationFactory : IMatlabPopulationFactory
    {
       private readonly IMatlabPopulationSettingsToPopulationSettingsMapper _populationSettingsMapper;
       private readonly IRandomPopulationFactory _randomPopulationFactory;
+      private readonly IOntogenyRepository _ontogenyRepository;
+      private readonly IMoleculeOntogenyVariabilityUpdater _ontogenyVariabilityUpdater;
+      private readonly IIndividualEnzymeFactory _individualEnzymeFactory;
 
       static MatlabPopulationFactory()
       {
          ApplicationStartup.Initialize();
       }
 
-      public MatlabPopulationFactory() : this(IoC.Resolve<IMatlabPopulationSettingsToPopulationSettingsMapper>(), IoC.Resolve<IRandomPopulationFactory>())
+      public MatlabPopulationFactory() : this(IoC.Resolve<IMatlabPopulationSettingsToPopulationSettingsMapper>(), IoC.Resolve<IRandomPopulationFactory>(),
+         IoC.Resolve<IOntogenyRepository>(), IoC.Resolve<IMoleculeOntogenyVariabilityUpdater>(), IoC.Resolve<IIndividualEnzymeFactory>())
       {
       }
 
-      public MatlabPopulationFactory(IMatlabPopulationSettingsToPopulationSettingsMapper populationSettingsMapper, IRandomPopulationFactory randomPopulationFactory)
+      public MatlabPopulationFactory(IMatlabPopulationSettingsToPopulationSettingsMapper populationSettingsMapper, IRandomPopulationFactory randomPopulationFactory,
+         IOntogenyRepository ontogenyRepository, IMoleculeOntogenyVariabilityUpdater ontogenyVariabilityUpdater, IIndividualEnzymeFactory individualEnzymeFactory)
       {
          _populationSettingsMapper = populationSettingsMapper;
          _randomPopulationFactory = randomPopulationFactory;
+         _ontogenyRepository = ontogenyRepository;
+         _ontogenyVariabilityUpdater = ontogenyVariabilityUpdater;
+         _individualEnzymeFactory = individualEnzymeFactory;
       }
 
-      public IParameterValueCache CreatePopulation(PopulationSettings matlabPopulationSettings, IEnumerable<string> moleculeNames)
+      public IParameterValueCache CreatePopulation(PopulationSettings matlabPopulationSettings, IEnumerable<MoleculeOntogeny> moleculeOntogenies)
       {
          var populationSettings = _populationSettingsMapper.MapFrom(matlabPopulationSettings);
          var population = _randomPopulationFactory.CreateFor(populationSettings, new CancellationToken()).Result;
+
+         foreach (var moleculeOntogeny in moleculeOntogenies)
+         {
+            var allOntogeniesForSpecies = _ontogenyRepository.AllFor(matlabPopulationSettings.Individual.OriginData.Species).ToList();
+            if (!allOntogeniesForSpecies.Any())
+               continue;
+
+            var ontogeny = allOntogeniesForSpecies.FindByName(moleculeOntogeny.Ontogeny);
```

Comment:
Shouldn't that be an error here? Somehow someones asks for an ontogeny that does not exist?

Reply:
looking good otherwise. If this is ok, you can merge